apps/otlp.go

```diff
@@ -44,10 +44,10 @@ func (r ReceiverOTLP) Type() string {
 
 func (ReceiverOTLP) gmpResourceProcessors() []otel.Component {
 	// Keep in sync with logic in confgenerator/prometheus.go
-	stmt := func(target, source string) string {
+	stmt := func(target, source, platform string) string {
 		// if cloud.provider == "gcp" && cloud.platform == "gcp_compute_engine"
-		return fmt.Sprintf(`set(%s, %s) where %s != nil and resource.attributes["cloud.platform"] == "gcp_compute_engine"`,
-			target, source, source)
+		return fmt.Sprintf(`set(%s, %s) where %s != nil and resource.attributes["cloud.platform"] == "%s"`,
+			target, source, source, platform)
 	}
 	return []otel.Component{
 		otel.GCPResourceDetector(false),
```

Comment:
I think you asked about this line of code. You need to replace `otel.GCPResourceDetector(false)` with `otel.ResourceTransform(pl.ResourceOverride.OTelResourceAttributes(), false)` when there's a resource override specified. (Effectively, the `ResourceOverride` is taking the place of the `GCPResourceDetector`)